src/arch/dotproductavx.cpp

```diff
@@ -16,8 +16,9 @@
 ///////////////////////////////////////////////////////////////////////
 
 #if !defined(__AVX__)
-#error Implementation only for AVX capable architectures
-#endif
+  //#error Implementation only for AVX capable architectures
+  int dotproductavx_empty_compilation_unit;
```

Comment:
Please don't define unused variables. They will trigger compiler warnings and be reported when searching for unused code. It is better to make an empty compilation unit.

Reply:
I have used systems in the past where empty compilation units give errors. I've seen such dummy variables used in other projects in the past, so I'm sure I'm not alone in this.

The dummy variable is a global, and therefore cannot be detected as unused by static analysis tools. In order for tools to spot it as unused, they'd actually have to do the compile and link, and even then they'd have to be very confident that it wasn't going to be referenced by a caller of Tesseract.

I will happily remove the variables if you want though, and we can put them back in if we hit such a system in future.

Reply:
The compiler will complain because each global symbol is expected to be declared in a header file.

Reply:
You can try to write `static void dummy(){}`.

Reply:
stweil: Good point. I'll do a revised commit that removes them, and we can worry about it again if a platform pops up where it's an issue.

egotpugin: static void dummy(void) {} would definitely get picked up by static analysis tools as being unused, though?

Reply:
Don't know. Maybe add some attribute. But, it's better than variable and seen in other projects.

Reply:
I am happy to remove the ints and either to add "static void dummy(void) {}" or not as you guys prefer.

If you could agree on the preferred approach, I'll gladly implement it. Thanks :)

Reply:
I clearly prefer the empty compilation unit.

Reply:
I'm fine with empty/function. But some compielrs warn about empty files.

Reply:
As long as they only warn, that's not a problem. It looks like we cannot avoid warnings from some tools, unless the build process skips those source files.